### PR TITLE
NO-ISSUE support using a local registry

### PR DESCRIPTION
**Pull request by michaellevy101**
Support using a local registry instead of pushing images directly to the minikube

### CODE REVIEW

**Review by flaper87 — COMMENTED** [review]
Love this change, I will let other folks that are more familiar with this part of the code the final approval.

File: debug.md [hunk 1]
```diff
@@ -32,11 +34,15 @@ You have two options to update the image:
         ```shell
         docker pull IMAGE_NAME
         k3d image import IMAGE_NAME
+   *  For a local registry
+      docker pull IMAGE_NAME 
+      docker tag REMOTE_IMAGE_NAME LOCAL_IMAGE_NAME
```
Comment (YuviGold) [inline debug.md:39]:
```suggestion
      docker pull REMOTE_IMAGE_NAME 
      docker tag REMOTE_IMAGE_NAME LOCAL_IMAGE_NAME  
```

File: Makefile [hunk 2]
```diff
@@ -146,6 +159,19 @@ update-debug-minimal:
 	docker build $(CONTAINER_BUILD_PARAMS) --build-arg SERVICE=$(SERVICE) --build-arg DEBUG_PORT=$(DEBUG_PORT) -f build/debug-image/Dockerfile.assisted-service-debug build/debug-image -t $(SERVICE)
 	rm -r build/debug-image
 
+update-image: $(UPDATE_IMAGE)
+
+_update-private-registry-image: update-image
```
Comment (YuviGold) [inline Makefile:164]: No need for a new target if it's an internal one and only calls to another target
```suggestion
_update-private-registry-image: $(UPDATE_IMAGE)
```

**Review by YuviGold — COMMENTED** [review]
Looks really good. I'm in favor of using an internal registry, that's a way better solution than relying on the minikube solution.

I still think it should be replaced with the current workflow. Otherwise, it won't be widely used by the contributors. Most of the contributors won't notice it unless it will require them to get to know the new workflow :)

File: Makefile (hunk 2 above)
Comment (michaellevy101) [reply Makefile:164]: This target is used by the programmer to update the image (like a public api)

**Comment by michaellevy101** [comment]
IMO using a **short** transition period is more elegant way to let the developers time to migrate their environment and let us deal with unexpected issues.
I hope most contributors will cooperate with this change

**Comment by YuviGold** [comment]
Take a look @filanov

File: debug.md [hunk 3]
```diff
@@ -14,13 +14,15 @@ The default remote debug port is 40000 but its configurable by setting the `DEBU
 export DEBUG_PORT=8765
 ```
 
-Compile the code with debug information, patch the image and push to your local k8s:
+Build the image and push to your local k8s:
```
Comment (filanov) [inline debug.md:17]: I think that this doc belongs under https://github.com/openshift/assisted-service/tree/master/docs/dev

**Comment by filanov** [comment]
I think that you need to add documentation explaining how it should be used.
It's not clear what I need to do in order to use `LOCAL_ASSISTED_ORG`. for example for minikube, i need to install an addon for other things I need to do ... 
IMO it can be helpful.

**Pull request merged**